### DISCUSSION TITLE
API rule: "allow only missing or empty"

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#1054] [Q&A] [answered]
I want to allow a field to be `''` or omitted (defaults to `''` on server) on record creation:

```ts
client.collection('myCollection').create({ myField: ''}) // ok - explicit ''
client.collection('myCollection').create({ }) // ok - server defaults to ''
client.collection('myCollection').create({ myField: 'oops'}) // validation error
```

Is there any way to express this with API rules? `myField=''` seems to demand that `myfield: ''` is passed in.

**Comment by ganigeorgiev** [comment, marked as answer]
`myField = ""` should work for the Create API rule if it is not marked "Nonempty".

**Reply by benallfree** [reply]
I have the API rule set to `message = ''` and it get this error when `message` is omitted from the `create` call:

```
[ERROR] create-instance:3 {
  "code": 400,
  "message": "Failed to create record.",
  "data": {
    "status": {
      "code": "validation_required",
      "message": "Missing required value."
    }
  }
}
```

<img width="679" alt="image" src="https://user-images.githubusercontent.com/1068356/202557233-4f88abc9-7c41-4635-ae1a-1ecbba342ef9.png">


<img width="689" alt="image" src="https://user-images.githubusercontent.com/1068356/202557132-42c464bb-a549-4525-8c91-82e92cf757a1.png">

**Reply by ganigeorgiev** [reply]
The above error message indicates that the `status` field has "Nonempty" enabled but I'm not sure why in your case it works if message is set because the field validators are independent from the API rules.

**Reply by benallfree** [reply]
Sorry, it seems to be working now. I'm not sure what changed. I'll update if I find it again.

**Reply by benallfree** [reply]
>  the status field has "Nonempty"

🤦  Totally my error
